lib/core/src/chain_swap.rs

```diff
@@ -1165,6 +1170,39 @@ impl ChainSwapHandler {
         }
     }
 
+    async fn fetch_incoming_swap_actual_payer_amount(&self, chain_swap: &ChainSwap) -> Result<u64> {
+        let swap_script = chain_swap.get_lockup_swap_script()?;
+        let address = swap_script
+            .as_bitcoin_script()?
+            .to_address(self.config.network.as_bitcoin_chain())
+            .map_err(|e| anyhow!("Failed to get swap script address {e:?}"))?;
+        let script_pubkey = address.script_pubkey();
+        let get_balance_res = self
+            .bitcoin_chain_service
+            .lock()
+            .await
+            .script_get_balance_with_retry(&script_pubkey, 10)
+            .await?;
+        debug!(
+            "Found user lockup balance for swap {}: {get_balance_res:?}",
+            chain_swap.id
+        );
+        let actual_payer_amount = match get_balance_res.confirmed > 0 {
```

Comment:
In case there is both confirmed and unconfirmed balance wouldn't it be risky to consider only the confirmed?
I am thinking of a case where the swapper now can redeem both balances.
Would it be safer to sum both balances here?

Reply:
Thanks for focusing on this issue. We should just take the first tx spending to the lockup script (the user lockup tx), just like we do in the recovery process. See [f233e49](https://github.com/breez/breez-sdk-liquid/pull/667/commits/f233e4935d11108a955967a5504efad629eef7f3). 

We should consider refactoring the code to reduce duplication of verification logic in here (swapper update handling in general) and in the recoverer. Not something for this PR of course.

Reply:
Agree